youtube_dl/extractor/ciscolive.py

```diff
@@ -0,0 +1,143 @@
+                'ext': 'mp4',
+            },
+        }
+    ]
+
+    # These appear to be constant across all Cisco Live presentations
+    # and are not tied to any user session or event
+    RAINFOCUS_API_URL = 'https://events.rainfocus.com/api/%s'
+    RAINFOCUS_APIPROFILEID = 'Na3vqYdAlJFSxhYTYQGuMbpafMqftalz'
+    RAINFOCUS_WIDGETID = 'n6l4Lo05R8fiy3RpUBm447dZN8uNWoye'
+    BRIGHTCOVE_URL_TEMPLATE = 'http://players.brightcove.net/5647924234001/SyK2FdqjM_default/index.html?videoId=%s'
+
+    def _parse_rf_item(self, rf_item):
+        ''' Parses metadata and passes to Brightcove extractor
+
+        '''
+        event_name = rf_item.get('eventName')
+        cl_id = rf_item.get('abbreviation')
+        title = rf_item.get('title')
+        description = clean_html(rf_item.get('abstract'))
+        presenter_name = try_get(rf_item, lambda x: x['participants'][0]['fullName'])
+        presenter_title = try_get(rf_item, lambda x: x['participants'][0]['jobTitle'])
+        pdf_url = try_get(rf_item, lambda x: x['files'][0]['url'])
+        bc_id = try_get(rf_item, lambda x: x['videos'][0]['url'])
+        bc_url = self.BRIGHTCOVE_URL_TEMPLATE % bc_id
+        duration = try_get(rf_item, lambda x: x['times'][0]['length'])
+        location = try_get(rf_item, lambda x: x['times'][0]['room'])
+
+        if duration:
+            duration = duration * 60
+
+        return {
+            '_type': 'url_transparent',
+            'url': bc_url,
+            'id': cl_id,
+            'title': '%s [%s]' % (title, cl_id),
+            'creator': '%s, %s' % (presenter_name, presenter_title),
+            'description': '%s\n\nVideo Player: %s\nSlide Deck: %s' % (description, bc_url, pdf_url),
+            'series': event_name,
+            'duration': duration,
+            'location': location,
+            'ie_key': 'BrightcoveNew',
+        }
+
+    def _check_bc_id_exists(self, rf_item):
+        ''' Checks for the existence of a Brightcove URL in a
+            RainFocus result item
+
+        '''
+        bc_id = try_get(rf_item, lambda x: x['videos'][0]['url'])
+        mobj = re.match(r'\d+', bc_id)
+        if mobj:
+            return rf_item
+
+    def _real_extract(self, url):
+        mobj = re.match(self._VALID_URL, url)
+        headers = {
+            'Origin': 'https://ciscolive.cisco.com',
+            'rfApiProfileId': self.RAINFOCUS_APIPROFILEID,
+            'rfWidgetId': self.RAINFOCUS_WIDGETID,
+            'Referer': url,
+        }
+
+        # Single session URL (single video)
+        if mobj.group('id'):
+            rf_id = mobj.group('id')
+            request = self.RAINFOCUS_API_URL % 'session'
+            data = compat_urllib_parse_urlencode({'id': rf_id})
+            rf_result = self._download_json(request, rf_id, data=data,
+                                            headers=headers)
+            rf_item = self._check_bc_id_exists(try_get(rf_result, lambda x: x['items'][0], dict))
```

Comment:
`try_get` is useless here.

Reply:
Without `try_get`, we'll get a `KeyError` or `IndexError` if/when the RainFocus API result changes schema. I've added my own `ExtractorError` in the latest commit, does that work?

Reply:
It does not matter here cause this data is mandatory (read coding conventions) and you can't proceed without it anyway. Do not raise `ExtractorError` on your own.